src/targets/gpu/compile_gen.cpp

```diff
@@ -169,6 +172,116 @@ bool preload::is_preloading() const
     return std::accumulate(args.begin(), args.end(), false, std::logical_or<>{});
 }
 
+static std::size_t integer_divide_ceil(std::size_t x, std::size_t y)
+{
+    return (x + y - std::size_t{1}) / y;
+}
+
+static std::size_t compute_tile_factor(std::size_t r, std::size_t max_size = 64)
+{
+    std::size_t n = 1;
+    auto factors  = make_array(2, 3, 5, 7, 11);
```

Comment:
Do we want any higher primes as part of the factors? 13, 17 and 19, 23 are also close (eg 23 is 11*2 + 1 ), but is there some sort of trade between higher primes are better tile factors?

Reply:
Actually I suppose smaller primes yield more tiles and thus easier to fit  without the risk of running out of memory? More a question on my end vs a design concern. I get what you're doing here, just curious as to why not have some additional searches

Reply:
I have not seen models using that high of prime factors in the dimensions. The important factor is that each wavefront is processing one row in the tile.

Reply:
Okay. That makes sense. Fair. I suppose until we hit a model with the higher dim, then leave them out.